app/scripts/controllers/membership.js

```diff
@@ -0,0 +1,304 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  .controller('MembershipController', [
+    '$filter',
+    '$location',
+    '$routeParams',
+    '$scope',
+    '$timeout',
+    '$uibModal',
+    'AuthorizationService',
+    'DataService',
+    'ProjectsService',
+    'MembershipService',
+    'RoleBindingsService',
+    'RolesService',
+    'LocalStorageUserStore',
+    function(
+      $filter,
+      $location,
+      $routeParams,
+      $scope,
+      $timeout,
+      $uibModal,
+      AuthorizationService,
+      DataService,
+      ProjectsService,
+      MembershipService,
+      RoleBindingsService,
+      RolesService,
+      LocalStorageUserStore) {
+
+      var requestContext;
+      var user = LocalStorageUserStore.getUser();
+      var projectName = $routeParams.project;
+      var startCase = $filter('startCase');
+      var formatKind = function(kindName) {
+        return startCase(kindName).toLowerCase();
+      };
+      var allRoles = [];
+
+      // NOTE: these could all be moved out into a strings service.
+      var messages = {
+        errorReason: _.template('Reason: "<%= httpErr %>"'),
+        notice: {
+          yourLastRole: _.template('Removing the role "<%= roleName %>" may completely remove your ability to see this project.')
+        },
+        warning: {
+          serviceAccount: _.template('Removing a system role granted to a service account may cause unexpected behavior.')
+        },
+        remove: {
+          areYouSure: {
+            subject: _.template('Are you sure you want to remove <strong><%= roleName %></strong> from the <%= kindName %> <strong><%= subjectName %></strong>?'),
+            self: _.template('Are you sure you want to remove <strong><%= roleName %></strong> from <strong><%= subjectName %></strong>?')
+          },
+          success: _.template('The role "<%= roleName %>" was removed from "<%= subjectName %>".'),
+          error: _.template('The role "<%= roleName %>" was not removed from "<%= subjectName %>".')
+        },
+        update: {
+          subject: {
+            success: _.template('The role "<%= roleName %>" was given to "<%= subjectName %>"'),
+            error: _.template('The role "<%= roleName %>" was not given to "<%= subjectName %>"')
+          }
+        }
+      };
+
+      // NOTE: alert service?
+      var showAlert = function(name, type, msg, detail, scope) {
+        scope = scope || $scope;
+        scope.alerts[name] = {
+          type: type,
+          message: msg
+        };
+        if(detail) {
+          scope.alerts[name].details = detail;
```

Comment:
why not just set details: detail in the map above, if detail is undefined the alert directive should handle that